doc/compiler/regular_expression/Regular_Expression.md

```diff
@@ -0,0 +1,41 @@
+# Regular expression
+
+A **regular expression** or **regex** is a concise way to describe matching or searching criteria for text strings. For more on regular expressions in general, see the [Wikipedia article](https://en.wikipedia.org/wiki/Regular_expression).
+
+Testarossa's `-Xjit` option uses regular expressions for many different debugging facilities. It should be noted that Testarossa implements a simplified regex matcher and it works a little differently compared to complete regex matchers. These regular expressions are not the same as those used by Perl or grep.
+
+## Syntax
+
+Regular expressions in `-Xjit` options are delimited by curly braces, and the braces are not considered part of the regular expression itself. Most characters in a regular expression match only themselves. For example, consider the following option:
+
+```
+-Xjit:limit={foo.method()V}
+```
+
+The regular expression here is "`foo.method()V`". This option will cause the jit only to compile the method whose signature exactly matches this string. The method names to search for are the ones after they have been mangled. The mangled method names are determined by whatever the internal mangling is for the compiler implementation. For example, in C++, a method `bar` in the class `Foo` that accepts an integer and retruns an integer will be mangled to something like `_ZN3Foo3barEi`. In Java, it would be `Foo.bar(I)I`. The mangling can be determined by running the following option:
```

Comment:
* Remove quotes (") around `foo.method()V`
* This option specifies that only the method whose signature exactly matches this string be compiled